test/scaling/scaling.jl

```diff
@@ -0,0 +1,52 @@
+module ScalingTests
+
+using Interpolations
+using Base.Test
+
+# Model linear interpolation of y = -3 + .5x by interpolating y=x
+# and then scaling to the new x range
+
+itp = interpolate(1:1.0:10, BSpline(Linear), OnGrid)
+
+sitp = scale(itp, -3:.5:1.5)
+
+for (x,y) in zip(-3:.05:1.5, 1:.1:10)
+    @test_approx_eq sitp[x] y
+end
+
+# Verify that it works in >1D, with different types of ranges
+
+gauss(phi, mu, sigma) = exp(-(phi-mu)^2 / (2sigma)^2)
```

Comment:
Do remember that if you choose a function that is "bi-quadratic" (i.e., `∂^3f/∂x^3 = 0` in each of its variables) and use the `InPlaceQ` boundary condition, then interpolation should match the function to within machine precision. By choosing something that agrees with the underlying assumptions of the interpolation model, we can build a more rigorous test.

But that can be tackled later, no need to hold up this excellent PR now.